src/macros.php

```diff
@@ -193,26 +193,30 @@
     /*
      * Splits a collection into sections grouped by a given key.
      *
-     * @param string $sectionBy
+     * @param mixed $key
+     * @param mixed $sectionKey
+     * @param mixed $itemsKey
+     * @param bool $preserveKeys
      *
      * @return \Illuminate\Support\Collection
      */
-    Collection::macro('sectionBy', function ($sectionByKey) {
-        $sectionKeyRetriever = $this->valueRetriever($sectionByKey);
+    Collection::macro('sectionBy', function ($key, $sectionKey = null, $itemsKey = 'items', $preserveKeys = false): Collection {
```

Comment:
@mdrost fyi You can just push more commits to your fork, they will show up in this PR.